src/classes/frDonation.cls

```diff
@@ -53,84 +54,133 @@ public class frDonation extends frModel implements frSyncable {
+                    o.put('AccountId', supporter.get('AccountId'));
                 }
             } else {
                 if(createLogRecord) frUtil.logRelationshipError(getFrType(), funraiseId, 
                                             frUtil.Entity.SUPPORTER, supporterFunraiseId);
             }
         }
-        
+        //connect campaign
         Boolean campaignMappingDisabled = request.containsKey(META_CAMPAIGN_MAPPING_KEY) ?
-            Boolean.valueOf(request.get(META_CAMPAIGN_MAPPING_KEY)): true;
+                Boolean.valueOf(request.get(META_CAMPAIGN_MAPPING_KEY)): true;
         if(!campaignMappingDisabled) {
             String campaignId = String.valueOf(request.get('campaignGoalId'));
             if (String.isNotBlank(campaignId)) {
                 List<Campaign> campaigns = [SELECT Id FROM Campaign WHERE fr_ID__c = :campaignId];
                 if (campaigns.size() > 0) {
-                    o.CampaignId = campaigns[0].Id;
+                    o.put('CampaignId', campaigns[0].Id);
                 } else {
                     frUtil.logRelationshipError(getFrType(), funraiseId, 
                                                 frUtil.Entity.CAMPAIGN, campaignId);
                 }
             }
         }
-        
+        //connect subscription
         String subscriptionId = String.valueOf(request.get('subscriptionId'));
         if (String.isNotBlank(subscriptionId)) {
-            List<Subscription__c> subscriptions = [SELECT Id FROM Subscription__c WHERE fr_ID__c = :subscriptionId];
+            List<SObject> subscriptions = Database.query('SELECT Id FROM ' + subsciptionObjName + ' WHERE fr_ID__c = :subscriptionId');
             if (subscriptions.size() > 0) {
-                o.Subscription__c = subscriptions.get(0).Id;
+                if(!frUtil.hasNPCobjects()) o.put('Subscription__c', subscriptions.get(0).get('Id'));
+                else o.put('GiftCommitmentId', subscriptions.get(0).get('Id'));
             } else {
                 frUtil.logRelationshipError(getFrType(), funraiseId, 
                                             frUtil.Entity.SUBSCRIPTION, subscriptionId);
             }
         }
-        Opportunity opp = this.o;
-        
-        
+
+        //connect pledge
         Boolean isPledge = Boolean.valueOf(request.get('pledge'));
-        Boolean alreadyMappedToPledge = [SELECT COUNT() FROM Opportunity 
-                                         WHERE fr_Id__c = :funraiseId AND Funraise_Pledge__c != null] > 0;
-        if(!isPledge && !alreadyMappedToPledge) {
-            Pledge__c pledge = frPledge.findActive(opp.fr_Donor__c, opp.CloseDate);
-            opp.Funraise_Pledge__c = pledge != null ? pledge.Id : null;
+        if(frUtil.hasNPCobjects()) {
+            List<SObject> alreadyMappedToPledgeList= Database.query('SELECT id FROM GiftTransaction WHERE fr_Id__c = :funraiseId AND Funraise_Pledge__c != null');
```

Comment:
It seems like this query is re-querying the record we already have in memory (based on the fr_Id__c value).  Can we update the query on line 74 to add in selecting the column Funraise_Pledge__c and that way we don't need to perform another query here, we can just check if Funraise_Pledge__c is populated on the `o` object